activerecord/test/cases/associations/belongs_to_associations_test.rb

```diff
@@ -1017,6 +1017,22 @@ def test_replace_counter_cache
     assert_equal 1, topic.replies_count
   end
 
+  def test_ignores_negative_counter_cache
+    post = Post.create!(title: "foo", body: "bar", comments_count: -1)
+
+    assert_queries(1) do
+      assert_predicate post.comments, :empty?
+    end
+
+    post.comments.create!(body: "Thank you")
+    post.update!(comments_count: -1)
+    post.reload
```

Comment:
As this test case is to check counter cache but line#1029 will reload record and counter_cache testing will be perfor med as expected. It should not reload to debug correctly.

Reply:
It is reloaded because otherwise the `comments` are staying loaded and just size on it is called, but we need to use counter cache column.